run.py

```diff
@@ -59,57 +69,59 @@
 print("Extracting Features...")
 # Retrieve db rows and store as a dataframe
 df = dbHelper.getRowsAsDf("SELECT * from clips where url = '{:}' ORDER BY start ASC".format(vidUrl))
+# Drop null columns with features to be populated
+feature_list = ['mfcc', 'amplitude', 'pitch', 'word', 'subjectivity', 'polarity', 'pred_excitement', 'pred_highlight_rf', 'pred_highlight_nn']
+df.drop(columns=feature_list, axis=1, inplace=True)
+
+# Feature extraction: add video title
+df['video_title'] = videoDownloader.getVideoTitle(vidUrl)
+print(videoDownloader.getVideoTitle(vidUrl))
 
 # Feature extraction: mfcc values
 print("Extracting mfcc...")
 mfccVals = mfccExtractor.getMfcc(librosaPath, intervals)
 df['mfcc'] = mfccVals
-dbHelper.updateColumn(df, 'mfcc')
 
 # Feature extraction: amp values
 print("Extracting amplitudes...")
 ampVals = ampExtractor.getAmp(librosaPath, intervals)
 df['amplitude'] = ampVals
-dbHelper.updateColumn(df, 'amplitude')
 
 # Feature extraction: pitch values
 print("Extracting pitches...")
 pitchVals = pitchExtractor.getPitch(audioPath, intervals)
 df['pitch'] = pitchVals
-dbHelper.updateColumn(df, 'pitch')
 
 # Feature extraction: speech to text
 print("Extracting speech 2 text data...")
 speech2text_df = speech2text.getText(audioPath, intervals)
-# this is ugly but it works
-df['word'] = speech2text_df['word'].tolist()
-df['subjectivity'] = speech2text_df['subjectivity'].tolist()
-df['polarity'] = speech2text_df['polarity'].tolist()
-dbHelper.updateColumn(df, 'word')
-dbHelper.updateColumn(df, 'subjectivity')
-dbHelper.updateColumn(df, 'polarity')
-
+df = pd.concat([df, speech2text_df], axis=1)
```

Comment:
I don't trust concat to behave nicely so I kinda prefer the old ugly way

Reply:
nope) concat can, and should be trusted. The bug I introduced was embarrassingly silly. we basically had ...

df1 <- | amplitude | start | end | word | subjectivity | polarity
df2_words <- | word | subjectivity | polarity

you have to drop `word`, `subjectivity`, `polarity` from df1 before using concat

Reply:
can we do this in a way without dropping? Seems a little backwards to drop all the columns then add them again. Also if we add new columns to the schema we'll have to duplicate the changes here which could get annoying

Reply:
Mmmm - I agree. I'll remove the `drop` upstream, and the schema we get from the table will be the one we enforce.